lib/spack/spack/test/entry_points.py

```diff
@@ -66,18 +66,8 @@ def entry_points(group=None):
 @pytest.fixture()
 def mock_entry_points(tmp_path, monkeypatch):
     entry_points = entry_points_factory(tmp_path)
-    try:
-        try:
-            import importlib.metadata as importlib_metadata  # type: ignore # novermin
-        except ImportError:
-            import importlib_metadata
-        monkeypatch.setattr(importlib_metadata, "entry_points", entry_points)
-    except ImportError:
-        try:
-            import pkg_resources  # type: ignore
-        except ImportError:
-            return
-        monkeypatch.setattr(pkg_resources, "iter_entry_points", entry_points)
+    import importlib.metadata # type: ignore # novermin
```

Comment:
This fails. Also don't mock functions you don't own, better to mock get_entrypoints.

Reply:
That's fine with me - but then `get_entry_points` will be untested...

Reply:
Yup, but it's run in various other places.